crypto/src/signature.rs

```diff
@@ -212,4 +244,37 @@ mod tests {
             true
         );
     }
+    #[test]
+    fn bls_agg_verify() {
+        // The number of signatures in aggregate
+        let num_sigs = 10;
+        let message_length = num_sigs * 64;
+
+        let rng = &mut XorShiftRng::from_seed([0x3dbe6259, 0x8d313d76, 0x3237db17, 0xe5bc0654]);
+
+        let msg = (0..message_length).map(|_| rng.gen()).collect::<Vec<u8>>();
+        let data: Vec<&[u8]> = (0..num_sigs).map(|x| &msg[x * 64..(x + 1) * 64]).collect();
+
+        let private_keys: Vec<PrivateKey> =
+            (0..num_sigs).map(|_| PrivateKey::generate(rng)).collect();
+        let public_keys: Vec<_> = (0..num_sigs)
+            .map(|x| private_keys[x].public_key().as_bytes())
+            .collect();
```

Comment:
```
        let public_keys: Vec<_> = private_keys
            .iter()
            .map(|x| x.public_key().as_bytes())
            .collect();
```

to remove need to iterate and index (defeats purpose of map to use x as an index into the vector)